### PR TITLE
PLAT-24227: Add to esearch->searchEntry ability to search on recycledAt

### PR DESCRIPTION
**Pull request by drorsou**

### CODE REVIEW

File: plugins/search/providers/elastic_search/lib/api/types/order/items/KalturaESearchEntryOrderByItem.php [hunk 1]
```diff
@@ -31,6 +31,7 @@ class KalturaESearchEntryOrderByItem extends KalturaESearchOrderByItem
         KalturaESearchEntryOrderByFieldName::VIEWS_LAST_7_DAYS => ESearchEntryOrderByFieldName::VIEWS_LAST_7_DAYS,
         KalturaESearchEntryOrderByFieldName::PLAYS_LAST_1_DAY => ESearchEntryOrderByFieldName::PLAYS_LAST_1_DAY,
         KalturaESearchEntryOrderByFieldName::VIEWS_LAST_1_DAY => ESearchEntryOrderByFieldName::VIEWS_LAST_1_DAY,
+		KalturaESearchEntryOrderByFieldName::RECYCLED_AT => ESearchEntryOrderByFieldName::RECYCLED_AT,
```
Comment (inbal-ben-david) [inline plugins/search/providers/elastic_search/lib/api/types/order/items/KalturaESearchEntryOrderByItem.php:34]: fix indentation

File: plugins/search/providers/elastic_search/lib/model/enum/ESearchEntryOrderByFieldName.php [hunk 2]
```diff
@@ -21,4 +21,5 @@ interface ESearchEntryOrderByFieldName extends BaseEnum
     const PLAYS_LAST_7_DAYS = 'plays_7days';
     const VIEWS_LAST_1_DAY = 'views_1day';
     const PLAYS_LAST_1_DAY = 'plays_1day';
+	const RECYCLED_AT = 'recycled_at';
```
Comment (inbal-ben-david) [inline plugins/search/providers/elastic_search/lib/model/enum/ESearchEntryOrderByFieldName.php:24]: fix indentation

File: plugins/search/providers/elastic_search/lib/model/items/ESearchEntryItem.php [hunk 3]
```diff
@@ -203,8 +205,7 @@ public function getSingleItemSearchQuery(&$entryQuery, $allowedSearchTypes, &$qu
 				KalturaLog::log("Undefined item type[" . $this->getItemType() . "]");
 		}
 
-		if ($this->getItemType() == ESearchItemType::EXACT_MATCH
-			&& in_array($this->getFieldName(), self::$ignoreDisplayInSearchFields)
+		if (in_array($this->getFieldName(), self::$ignoreDisplayInSearchFields)
```
Comment (inbal-ben-david) [inline plugins/search/providers/elastic_search/lib/model/items/ESearchEntryItem.php:208]: why you removed the exact match type condition?

File: plugins/search/providers/elastic_search/lib/model/search/kEntrySearch.php [hunk 4]
```diff
@@ -23,7 +23,10 @@ protected function handleDisplayInSearch()
             return;
 
         $displayInSearchQuery = $this->queryAttributes->getQueryFilterAttributes()->getDisplayInSearchFilter();
-        $this->mainBoolQuery->addToFilter($displayInSearchQuery);
+		if ($displayInSearchQuery)
```
Comment (inbal-ben-david) [inline plugins/search/providers/elastic_search/lib/model/search/kEntrySearch.php:26]: fix indentation

File: plugins/search/providers/elastic_search/lib/model/items/ESearchEntryItem.php (hunk 3 above)
Comment (drorsou) [reply plugins/search/providers/elastic_search/lib/model/items/ESearchEntryItem.php:208]: The item type can now also be 'RANGE', we want to retrieve recycled entries when using 'recycled_at'

**Review by inbal-ben-david — APPROVED** [review]

**Pull request merged**